File: virtual time capsule/frontend/virtual-time-capsule-ui/src/app/dtos/capsule-response.dto.ts
import { CapsuleStatus } from '../enums/capsule-status.enum';
import { GoalDto } from './goal.dto';
import { MemoryDto } from './memory.dto';
import { UserProfileDto } from './user-profile.dto';

export interface CapsuleResponseDto {
  id: number;
  capsuleName: string;
  status: CapsuleStatus;
  creationDate: string;
  lockDate: string | null;
  openDateTime: string | null;
  creator: UserProfileDto;
  goalId: number | null;
  memories?: MemoryDto[];
  isShared: boolean;
  sharedWithUsers?: UserProfileDto[];
  readyParticipantsCount?: number;
  totalParticipantsCount?: number;
  isCurrentUserReadyToClose?: boolean;
}

File: virtual time capsule/backend/src/main/java/com/java/web/virtual/time/capsule/dto/CapsuleResponseDto.java
package com.java.web.virtual.time.capsule.dto;

import com.java.web.virtual.time.capsule.enums.CapsuleStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CapsuleResponseDto {
    private Long id;
    private String capsuleName;
    private CapsuleStatus status;
    private String creationDate;
    private String lockDate;
    private String openDateTime;
    private UserProfileDto creator;
    private Long goalId;
    private Set<MemoryDto> memories;
    private Boolean isShared;
    private Set<UserProfileDto> sharedWithUsers; 
    private Integer readyParticipantsCount; 
    private Integer totalParticipantsCount; 
    private Boolean isCurrentUserReadyToClose; 
}
